#!/usr/bin/env python
PACKAGE = "geometric_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# geometric_controller

gen.add("max_acc",    double_t,    0, "Max Feedback Acceleration", 0.1,  0.1, 8.0)
gen.add("Kp_x", double_t, 0, "Proportional gain for X position error", 1.0,  0.10, 10.0)
gen.add("Kp_y", double_t, 0, "Proportional gain for Y position error", 1.0, 0.1, 10.0)
gen.add("Kp_z", double_t, 0, "Proportional gain for Z position error", 1.0, 0.1, 10.0)
gen.add("Kv_x", double_t, 0, "Proportional gain for X velocity error", 2.0,  0.0, 10.0)
gen.add("Kv_y", double_t, 0, "Proportional gain for Y velocity error", 2.0, 0.0, 10.0)
gen.add("Kv_z", double_t, 0, "Proportional gain for Z velocity error", 2.0, 0.0, 10.0)
gen.add("Ki_x", double_t, 0, "Integral gain for X position error", 0.1,  0.0, 10.0)
gen.add("Ki_y", double_t, 0, "Integral gain for Y position error", 0.1, 0.0, 10.0)
gen.add("Ki_z", double_t, 0, "Integral gain for Z position error", 0.1, 0.0, 10.0)
exit(gen.generate(PACKAGE, "geometric_controller", "GeometricController"))
